### PR TITLE
[acceptance.pm] add oldlog folder and add code to create junit report

### PR DESCRIPTION
**Pull request by SeroSun**
1. add oldlog folder 
2. add code to create junit report
3. modify waiting time from 2 hours to 5 hours for a whole round of stress validation test.

### CODE REVIEW

File: tests/qa_automation/acceptance.pm [hunk 1]
```diff
@@ -65,8 +65,25 @@ sub run() {
     # QA DB upload happens for each module
     # output the failed tests to serial console
     type_string "export PS1=#\n";
+    # qa_testset_automation deleted oldlog folders, so we add it back to adopt it
+    assert_script_run "mkdir /var/log/qa/oldlogs; cd /var/log/qa/oldlogs; for i in /var/log/qaset/log/*.bz2; do tar -xjvf \$i; done";
```
Comment (bmwiedemann) [inline tests/qa_automation/acceptance.pm]: i is normally used for integers, so it is a bit misleading here. better f or file

Comment (coolo) [inline tests/qa_automation/acceptance.pm]: Not sure how you define "normally", but I use i in shell loops almost exclusivly:

<pre>
coolo@gertrude#os-autoinst-distri-opensuse>history | grep "for i" | wc -l
163
coolo@gertrude#os-autoinst-distri-opensuse>history | grep "for f" | wc -l
1
</pre>

Comment (coolo) [inline tests/qa_automation/acceptance.pm]: (the 1 is actually the history call itself :)

File: tests/qa_automation/acceptance.pm [hunk 2]
```diff
@@ -65,8 +65,25 @@ sub run() {
...
     type_string "find /var/log/qa/oldlogs/ -name test_results | xargs grep -l -B1 ^1 | tee -a /dev/$serialdev\n";
     assert_screen 'no_output_from_qa_find';
+    # create a junit report
```
Comment (coolo) [inline tests/qa_automation/acceptance.pm]: I don't like this loop at all. All that's different between the ifs is argument to -t, so better calculate that variable in an if-else and then have one line to run the command using the variable.

**Comment by coolo** [comment]
do you plan to improve this any time soon?

**Comment by coolo** [comment]
ping!

**Comment by SeroSun** [comment]
I'm sorry! I just notice your reply, I guest my mail box have some problem with notification in github. I'll update it soon.

File: tests/qa_automation/acceptance.pm [hunk 3]
```diff
@@ -64,9 +68,29 @@ sub run() {
...
+    # qa_testset_automation remove the oldlogs, we create it back here for junit collecting information
+    assert_script_run "mkdir /var/log/qa/oldlogs; cd /var/log/qa/oldlogs; for i in /var/log/qaset/log/*.bz2; do tar -xjvf \$i; done";
     type_string "find /var/log/qa/oldlogs/ -name test_results | xargs grep -l -B1 ^1 | tee -a /dev/$serialdev\n";
     assert_screen 'no_output_from_qa_find';
+
+    # test junit
+    my $testset = get_var('QA_TESTSET');
+    if ($testset == "acceptance") { 
+        my $type = "stress_validation";
+    }
```
Comment (coolo) [inline tests/qa_automation/acceptance.pm]: my creates a scope local variable. This won't fly - and is obviously not tested

File: tests/qa_automation/acceptance.pm [hunk 4]
```diff
@@ -64,9 +68,29 @@ sub run() {
...
+    elsif  ($testset == "regression") {
+        my $type = "user_regression";
+    }
+    elsif  ($testset == "kernel") {
+        my $type = "kernel_regression";
+    }
+    else {
+        my $type = "all";
+    }
+    assert_script_run "/usr/share/qa/qaset/bin/junitxml_generator.py -t " . $type . " -l /var/log/qaset/runs/ -s /var/log/qaset/submission/ -o /tmp/junit.xml";
```
Comment (coolo) [inline tests/qa_automation/acceptance.pm]: there is no need to concat variables like this. You can just use "-t $type" - perl expands variables in strings

**Comment by coolo** [comment]
now you add even more code duplication? Why would you do that?

**Comment by coolo** [comment]
superseded by https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/809

**Pull request closed without merging**